### PR TITLE
Bugfix: Fix mobile sync & Security options

### LINKED ISSUE #896: sync freeze
I am on this screen for >10 mins now - think it somehow hangs.

image

the spinner is turning though. Not much in the logs unfortunately
Btw.: where can I find details about the sync protocol?

### LINKED ISSUE #985: Fingerprint unlock with fingerprint unlocking disabled!
1. Install MetaMask app from Play Store.
2. Sync with desktop browser extension.
3. Go to security settings in app and ensure Fingerprint unlock is _disabled_.
4. Close MetaMask app.
5. Open MetaMask app.
--> Notice that you are prompted for a fingerprint to unlock the app!

This is a pretty big security issue for me as biometrics alone are _NOT_ secure.  While I have them setup for unlocking my phone, all of my financial apps are secured behind _something I know_, not something I have since getting into my phone in the first place requires something I have (biometrics + device).

### PR DESCRIPTION
**Pull request by brunobar79**
This PR addresses three separate issues related to sync & security / biometrics

1) I was able to reproduce the infinite spinner and at the same time found the root cause of it.

When one request failed, the retry logic kept two listeners active, which causes all the messages to arrive twice.  That breaks the json structure while rebuilding the state, throwing an exception on JSON.parse

This PR updates the way we handle the listener removal based on the pubnub docs and also bumps `pubnub` dep to its latest version, which was different to the one that we were using in the extension and I'm updating on a separate PR.

In order to fix the issue with older versions of the extension where the same messages can still be sent twice I've added a check to make sure if it's repeated to be ignored.

Fixes #896

2)  Now users will get prompted to enable biometrics while syncing with the extension.
Additionally fixes a bug with the flag `biometryChoiceDisabled` which was being reset every time the login screen was shown 

Fixes #985 and also #927 

3) Added the ability to enable / disable  device PIN / Passcode  separately from biometrics,  allowing users to use any of the 4 ways of logging in:
- password
- password + remember me
- device PIN / PASSCODE
- biometrics

If you disable biometrics it will fallback to device PIN, if you disable device PIN it will default to password.

### CODE REVIEW

File: app/components/Views/EnterPasswordSimple/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+	onPressConfirm = async () => {
+		if (this.state.loading) return;
+		let error = null;
+		if (this.state.password.length < 8) {
+			error = strings('choose_password.password_length_error');
+		}
+
+		if (error) {
+			Alert.alert('Error', error);
```
Comment (estebanmino) [inline app/components/Views/EnterPasswordSimple/index.js]: locale? and why not add this to the error check 4 lines above?

**Review by estebanmino — APPROVED** [review]
💯 💯 nice pr, code lgtm

File: app/components/Views/EnterPasswordSimple/index.js (hunk 1 above)
Comment (brunobar79) [reply app/components/Views/EnterPasswordSimple/index.js]: Agreed. Took that from another view and didn't refactor as much as I should.

**Comment by ibrahimtaveras00** [comment]
**Issue 1:**
_happens on iOS_

When saying no to touch ID and then killing/relaunching app, I’m prompted with touch ID 

Steps:
-When prompted to use touch ID say no
-go to settings > privacy ensure both touch ID and passcode are disabled 
-kill app and relaunch notice you’re prompted for touch ID (or faceID if enabled) 

**Issue 2:** 
_happens on iOS_

I enable touch ID via settings, then log out, I still see remember me and then once logging back in, touch ID is disabled 

Steps:
-When prompted to use touch ID say no
-go to settings > privacy and enable touch ID
-log out, and notice how it still says remember me
-log back in and go to settings and notice how touch ID is now disabled  

Same thing happens if I enable touch ID in settings, kill app, relaunch, then log out 

**Issue 3:**
_happens on both iOS and Android_

When on a device that doesn't support biometrics, but does support pin/passcode I get an error

*Note: this happens with a device that doesn't have pin enabled as well*

Seen here = http://recordit.co/CSfy5TWEQj

image

image

**Issue 4:**
_Happens on Android_

When I try the import seed phrase flow, and toggle OFF the fingerprint option then continue onwards, before being shown the opt in view, I’m prompted to enter my pin passcode where it says “Please authenticate in order to use MetaMask”

Steps:
-Import via seed phrase but toggle OFF the fingerprint option before continuing

**Comment by ibrahimtaveras00** [comment]
Fixes look good, QA Passed 👍

**Pull request merged**